app/javascript/crayons/formElements/DateRangePicker/DateRangePicker.jsx

```diff
@@ -86,6 +95,62 @@ const MonthYearPicker = ({
   );
 };
 
+/**
+ * Renders preset date ranges as 'quick select' buttons, if the range falls within the permitted dates
+ *
+ * @param {[string]} presetRanges The preset range names requested
+ * @param {Object} earliestMoment Moment object representing earliest permitted date
+ * @param {Object} latestMoment Moment object representing latest permitted date
+ * @param {Function} onPresetSelected Callback which will receive start and end dates of selected preset
+ * @param {Object} today Moment object representing today's date
+ */
+const PresetDateRangeOptions = ({
+  presetRanges = [],
+  earliestMoment,
+  latestMoment,
+  onPresetSelected,
+  today,
+}) => {
+  const presetsWithinPermittedDates = presetRanges.filter((rangeName) => {
+    const { start, end } = getDateRangeStartAndEndDates({
+      today,
+      dateRangeName: rangeName,
+    });
+
+    return (
```

Comment:
I had to read this twice, so I'm adding reading notes:

`start` is the user given value, `earliestMoment` is the allowed range by app logic.  `end` is the user given value, `latestMoment` is the allowed range by app logic.

Do we need to enforce that `end` is after `start`?

If so, I believe we can do this:

`start.isSameOrBefore(end) && start.isSameOrAfter(earliestMoment) && end.isSameOrBefore(latestMoment)`

Reply:
We don't _really_ need to worry about `end` being after `start`, as the end/start values come from our own pre-defined/hard-coded date ranges. The `earliestMoment` and `latestMoment` however come from whichever view is implementing the date range picker, so the purpose of this function is largely for ease of developer experience (otherwise whenever we use this component we'd have to do some calculations to determine which presets are appropriate for the given context/user).

But yeah you're right, we can simplify this 👍  I'll also look to add some comments here to make things more readable